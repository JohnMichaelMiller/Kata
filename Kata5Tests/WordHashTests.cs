using Microsoft.VisualStudio.TestTools.UnitTesting;
using DifferentialAbstraction;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ApprovalTests;
using ApprovalTests.Reporters;
using ApprovalUtilities.Utilities;

namespace DifferentialAbstraction.Tests
{

    [TestClass()]
    [UseReporter(typeof(DiffReporter))]
    public class WordHashTests
    {
        const string fileName = @"C:\Users\John\Source\Repos\Kata\Kata5\wordlist.txt";

        [TestMethod()]
        [UseReporter(typeof(ClipboardReporter), typeof(DiffReporter))]
        public void WordHashesTest()
        {
            var words = new WordHash(fileName);
            Approvals.VerifyAll(words, d => "Key {0}, {1}".FormatWith(d.Key, WordHash.WordToString(d)));
        }

        [TestMethod()]
        public void WordHashctor0Test()
        {
            WordHash wh = new WordHash();
            Assert.AreEqual(wh.Count,0);
        }

        [TestMethod()]
        public void WordHashctor1Test()
        {
            WordHash wh = new WordHash(new string[] { "Cat", "Dog" });
            Approvals.VerifyAll(wh, d => "Key {0}, {1}".FormatWith(d.Key, WordHash.WordToString(d)));
        }
        [TestMethod()]
        public void WordHashctor2Test()
        {
            WordHash wh = new WordHash(fileName);
            Approvals.VerifyAll(wh, d => "Key {0}, {1}".FormatWith(d.Key, WordHash.WordToString(d)));
        }

        [TestMethod()]
        public void GetWordsFromFileTest()
        {
            WordHash wh = new WordHash();
            wh.GetWordsFromFile(fileName);
            Approvals.VerifyAll(wh, d => "Key {0}, {1}".FormatWith(d.Key, WordHash.WordToString(d)));
        }

        [TestMethod()]
        public void AddWordsTest()
        {
            WordHash wh = new WordHash();
            wh.AddWords(new string[] { "Cat", "Dog" });
            Approvals.VerifyAll(wh, d => "Key {0}, {1}".FormatWith(d.Key, WordHash.WordToString(d)));
        }

        [TestMethod()]
        public void GetHashTest()
        {
            Approvals.Verify(WordHash.HashToString(WordHash.GetHash("Test")));
        }

        [TestMethod()]
        public void WordToStringTest()
        {
            KeyValuePair<string, int> word = new KeyValuePair<string, int>("Test", 128);
            Approvals.Verify(WordHash.WordToString(word));
        }

        [TestMethod()]
        public void HashToStringTest()
        {
            Approvals.Verify(WordHash.HashToString(new BitArray(new int[] {1024})));
        }
    }
}