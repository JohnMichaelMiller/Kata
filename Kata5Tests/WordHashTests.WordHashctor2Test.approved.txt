Key A, A:-1684705346
Key A'asia, A'asia:1367389317
Key A's, A's:-719963608
Key AA, AA:1780295581
Key AA's, AA's:718935806
Key AAA, AAA:395669136
Key AAM, AAM:-316791584
Key AB, AB:1780098973
Key AB's, AB's:718608126
Key ABA, ABA:395406992
Key ABC, ABC:751637352
Key ABC's, ABC's:-987100334
